tools/gen.sh

```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
```

Comment:
I think it's better if the file name were more descriptional.
I mean rename to `gen-xxxx.sh`.
I guess that you are thinking about adding more functionality in this script, so how about `gen-code.sh` or something.

Reply:
renamed to `gen-code.sh`